lib/Mail/Milter/Authentication/Resolver.pm

```diff
@@ -21,19 +21,43 @@ to Net::DNS::Resolver
         my $self = $class->SUPER::new( @_ );
         weaken($args{_handler});
         $self->{ _handler } = $args{_handler};
+        $self->{ _timedout } = {};
         return $self;
     }
 }
 
+sub clear_error_cache {
+    my $self = shift;
+    $self->{ _timedout } = {};
+    return;
+}
+
 sub _do { ## no critic
     my $self = shift;
     my $what = shift;
+
     my $handler = $self->{_handler};
     my $config = $handler->config();
     my $timeout = $config->{'dns_timeout'};
 
     my $return;
 
+    my $org_domain = $_[0];
+    if ( $handler->is_handler_loaded( 'DMARC' ) ) {
+        my $dmarc_object = $handler->get_handler('DMARC')->get_dmarc_object();
+        $org_domain = eval{ $dmarc_object->get_organizational_domain( $org_domain ) };
+        $handler->handle_exception( $@ );
+    }
+
+    # If we have a 'cached' timeout for this org domain then return
+    if ( exists( $self->{ _timedout }->{ $org_domain } ) ) {
```

Comment:
You don't need `exists` here. It's not protecting against anything unless you plan to use non-true values to mean something timed out or not.